<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="6" total="8" passed="1" failed="1" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2024-01-21T18:47:40 EET" name="All Test Suite" finished-at="2024-01-21T18:47:55 EET" duration-ms="14952">
    <groups>
    </groups>
    <test started-at="2024-01-21T18:47:40 EET" name="login_with_valid_data" finished-at="2024-01-21T18:47:48 EET" duration-ms="8539">
      <class name="testcases.T01_login">
        <test-method is-config="true" signature="setupdriver(java.lang.String)[pri:0, instance:testcases.T01_login@543588e6]" started-at="2024-01-21T18:47:40 EET" name="setupdriver" finished-at="2024-01-21T18:47:46 EET" duration-ms="5717" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setupdriver -->
        <test-method signature="login_with_valid_data()[pri:1, instance:testcases.T01_login@543588e6]" started-at="2024-01-21T18:47:46 EET" name="login_with_valid_data" description="login with valid username and password " finished-at="2024-01-21T18:47:48 EET" duration-ms="2235" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- login_with_valid_data -->
        <test-method is-config="true" signature="getResult(org.testng.ITestResult)[pri:0, instance:testcases.T01_login@543588e6]" started-at="2024-01-21T18:47:48 EET" name="getResult" finished-at="2024-01-21T18:47:48 EET" duration-ms="18" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=login_with_valid_data status=SUCCESS method=T01_login.login_with_valid_data()[pri:1, instance:testcases.T01_login@543588e6] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getResult -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:testcases.T01_login@543588e6]" started-at="2024-01-21T18:47:48 EET" name="teardown" finished-at="2024-01-21T18:47:48 EET" duration-ms="297" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
      </class> <!-- testcases.T01_login -->
    </test> <!-- login_with_valid_data -->
    <test started-at="2024-01-21T18:47:49 EET" name="add_to_cart_with_invalid" finished-at="2024-01-21T18:47:55 EET" duration-ms="6175">
      <class name="testcases.T02_add_to_cart">
        <test-method is-config="true" signature="defineSuiteElements()[pri:0, instance:testcases.T02_add_to_cart@82de64a]" started-at="2024-01-21T18:47:39 EET" name="defineSuiteElements" finished-at="2024-01-21T18:47:40 EET" duration-ms="755" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- defineSuiteElements -->
        <test-method is-config="true" signature="setupdriver(java.lang.String)[pri:0, instance:testcases.T02_add_to_cart@82de64a]" started-at="2024-01-21T18:47:49 EET" name="setupdriver" finished-at="2024-01-21T18:47:53 EET" duration-ms="4024" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setupdriver -->
        <test-method signature="add_to_cart_invalid()[pri:1, instance:testcases.T02_add_to_cart@82de64a]" started-at="2024-01-21T18:47:53 EET" name="add_to_cart_invalid" description="add to cart with invalid scenario" finished-at="2024-01-21T18:47:54 EET" duration-ms="1650" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[The following asserts failed:
expected [remove] but found [Add to cart]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: The following asserts failed:
expected [remove] but found [Add to cart]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
at testcases.T02_add_to_cart.add_to_cart_invalid(T02_add_to_cart.java:23)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:136)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:658)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:219)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:923)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:192)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:808)
at org.testng.TestRunner.run(TestRunner.java:603)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
at org.testng.SuiteRunner.run(SuiteRunner.java:326)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
at org.testng.TestNG.runSuites(TestNG.java:1092)
at org.testng.TestNG.run(TestNG.java:1060)
at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:65)
at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:105)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- add_to_cart_invalid -->
        <test-method is-config="true" signature="getResult(org.testng.ITestResult)[pri:0, instance:testcases.T02_add_to_cart@82de64a]" started-at="2024-01-21T18:47:55 EET" name="getResult" finished-at="2024-01-21T18:47:55 EET" duration-ms="1" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[[TestResult name=add_to_cart_invalid status=FAILURE method=T02_add_to_cart.add_to_cart_invalid()[pri:1, instance:testcases.T02_add_to_cart@82de64a] output={null}]]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getResult -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:testcases.T02_add_to_cart@82de64a]" started-at="2024-01-21T18:47:55 EET" name="teardown" finished-at="2024-01-21T18:47:55 EET" duration-ms="205" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:testcases.T02_add_to_cart@82de64a]" started-at="2024-01-21T18:47:55 EET" name="tearDown" finished-at="2024-01-21T18:47:56 EET" duration-ms="1059" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- testcases.T02_add_to_cart -->
    </test> <!-- add_to_cart_with_invalid -->
  </suite> <!-- All Test Suite -->
</testng-results>
